packages/contracts/contracts/extensions/CompliantForwarder/CompliantForwarder.sol

```diff
@@ -0,0 +1,101 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity 0.4.24;
+pragma experimental ABIEncoderV2;
+
+import "../../protocol/Exchange/interfaces/IExchange.sol";
+import "../../tokens/ERC721Token/IERC721Token.sol";
+import "../../utils/LibBytes/LibBytes.sol";
+
+contract CompliantForwarder {
+
+    using LibBytes for bytes;
+
+    bytes4 constant internal EXCHANGE_FILL_ORDER_SELECTOR = bytes4(keccak256("fillOrder((address,address,address,address,uint256,uint256,uint256,uint256,uint256,uint256,bytes,bytes),uint256,bytes)"));
+    IExchange internal EXCHANGE;
+    IERC721Token internal COMPLIANCE_TOKEN;
+
+    constructor(address exchange, address complianceToken)
+        public
+    {
+        EXCHANGE = IExchange(exchange);
+        COMPLIANCE_TOKEN = IERC721Token(complianceToken);
+    }
+
+    function fillOrder(
```

Comment:
This uses the `executeTransaction` interface. Another option is to use the `fillOrder` interface, then ABI-encode within this function. Did it this way to save gas. Shouldn't be a huge deal if most clients use the TS wrapper we write.

Reply:
I think it would be valuable to pass in the actual `Order` and `takerAssetFillAmount`, validate them, and then ABI encode the params before passing them into executeTransaction. Even though this would be less gas efficient than what we're currently doing, this is much more extensible to other wrapper functions.